java/core/src/main/java/co/worklytics/psoxy/impl/SanitizerImpl.java

```diff
@@ -33,9 +35,18 @@
 import java.util.stream.Collectors;
 
 @Log
-@RequiredArgsConstructor
 public class SanitizerImpl implements Sanitizer {
 
+    @AssistedInject
+    public SanitizerImpl(HashUtils hashUtils, @Assisted Options options) {
+        this.options = options;
```

Comment:
why doesn't this assign `hashUtils`? does that happen magically in `AssistedInject`?

Reply:
yes. With `Assisted` you just tell the factory the dependencies that the caller must satisfy. Under the hood this is just like passing "options" to a `@Provides` method.